src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/Validation/MandatoryLocalizationValidationFactory.cs

```diff
@@ -0,0 +1,88 @@
+
+        internal class MandatoryLocalizationValidation : ITemplateValidator
+        {
+            internal MandatoryLocalizationValidation(ITemplateValidatorFactory factory)
+            {
+                Factory = factory;
+            }
+
+            public ITemplateValidatorFactory Factory { get; }
+
+            public void ValidateTemplate(ITemplateValidationInfo templateValidationInfo)
+            {
+                foreach (KeyValuePair<CultureInfo, TemplateLocalizationInfo> localization in templateValidationInfo.Localizations)
+                {
+                    LocalizationModel locModel = localization.Value.Model;
+                    int unusedPostActionLocs = locModel.PostActions.Count;
+                    foreach (PostActionModel postAction in templateValidationInfo.ConfigModel.PostActionModels)
+                    {
+                        if (postAction.Id == null || !locModel.PostActions.TryGetValue(postAction.Id, out PostActionLocalizationModel postActionLocModel))
+                        {
+                            // Post action with no localization model.
+                            continue;
+                        }
+
+                        unusedPostActionLocs--;
+
+                        // Validate manual instructions.
+                        bool instructionUsesDefaultKey = postAction.ManualInstructionInfo.Count == 1 && postAction.ManualInstructionInfo[0].Id == null &&
+                            postActionLocModel.Instructions.ContainsKey(PostActionModel.DefaultIdForSingleManualInstruction);
+                        if (instructionUsesDefaultKey)
+                        {
+                            // Just one manual instruction using the default key. No issues. Continue.
+                            continue;
+                        }
+
+                        int unusedManualInstructionLocs = postActionLocModel.Instructions.Count;
+                        foreach (var instruction in postAction.ManualInstructionInfo)
+                        {
+                            if (instruction.Id != null && postActionLocModel.Instructions.ContainsKey(instruction.Id))
+                            {
+                                unusedManualInstructionLocs--;
+                            }
+                        }
+
+                        if (unusedManualInstructionLocs > 0)
+                        {
+                            // Localizations provide more translations than the number of manual instructions we have.
+                            string excessInstructionLocalizationIds = string.Join(
+                                ", ",
+                                postActionLocModel.Instructions.Keys.Where(k => !postAction.ManualInstructionInfo.Any(i => i.Id == k)));
```

Comment:
is it possible that we have a lot of concatenations here?

Reply:
Not sure. This is existing code (now in https://github.com/dotnet/templating/blob/4fe046dbc4f7d079892750fa1b896c5a6356cd6e/src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/RunnableProjectConfig.cs#L306) and let's keep it as is for now.